app/components/sign-up/button.js

```diff
@@ -0,0 +1,26 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class ButtonComponent extends Component {
+  @action
+  buttonClickHandler() {
+    const { state, clickHandler, disabled } = this.args;
+
+    switch (state) {
+      case 'get-started':
+        !disabled ? clickHandler('firstName') : '';
```

Comment:
Can we extract this out in a function and call it with apt args as required?
This line seems to be redundant (with different params) in the cases below